### ISSUE TITLE
PiPico Board screw hole

### ISSUE THREAD

**Issue by carneeki** [terjeio/GRBL_MPG_DRO_BoosterPack#5]
Hi Terje,

In the PCB layout the J7 (controller) connector obscures a 3mm mounting hole (see first and second picture)
![image](https://user-images.githubusercontent.com/481448/148304419-754979d0-14d4-4f16-bedd-0a3763013f70.png)

![image](https://user-images.githubusercontent.com/481448/148304009-62e5e884-c907-438c-a3e0-56f113e52bb8.png)

By rotating the VSUP header, and moving U3 further to the right, room can be made for both the screw hole and J7 (see third picture)
![image](https://user-images.githubusercontent.com/481448/148303807-16755f08-65ff-4513-8cba-6ecf977327d5.png)

Also suggest using a 3.2mm hole; as it permits a #10-32 screw for those in the US.

I am hesitant to send a PR as I upgraded to KiCAD 6 and the files are not backward compatible (only forward compatible).

**Comment by terjeio** [comment]
It is likely that I will change the D socket to a smaller version or replacing it with different connector(s). The latest version of the firmware I am working on can work with only four wires, or soon maybe even two (for power, with wireless transmission of data) or none (if battery operated).

What is more interesting is a discussion about how to connect to the controller and which kind of enclosures could be used. From that a more appropriate PCB design can be made.

FYI I made a 5x5 keypad yesterday for testing, and I have more than 30 commands assigned already. It seems like this can be a powerful/flexible platform to build upon...

**Comment by carneeki** [comment]
Funny you mention enclosures, I am in the middle of designing one now that a friend offered to 3d print for me. My constraints are a 180x180 bed area.

![image](https://user-images.githubusercontent.com/481448/148541308-a21dcf2a-d363-4cc6-9132-efcd42ca5aa1.png)

I admit to going overkill with the screws, but that was an aesthetic choice ;)

![image](https://user-images.githubusercontent.com/481448/148541505-573a84c2-f3b0-4734-9903-9dfd0a0f44b9.png)

The connector will likely be a CX20 or CX24 style connectors with a short pigtail from connector to board. (if you opt for 0.1" headers, that would permit a wide variety of connection options, including many cable to board connectors). CX24 permits up to 26 pins in and out of the enclosure (3A per pin) and I have found them quite cheap on AliExpress. Currently thinking:

- 15 pins for the MPG/DRO board to main controlbox (currently Pi3 + Grbl, stepper drivers, 5V, 12V and 48V PSUs),
- 2 pins for 5V/12V power and I have a number of spare lines I can use in the future.
- 9 pins free for future use

I haven't put a display or buttons in yet; one idea that I thought about was to use buttons next to the display similar to what many oscilloscopes offer. The buttons can be assigned different tasks depending on the screen you are looking at. Here is a spectrum analyser I've used, the 5 buttons below the screen are redefineable "soft" buttons:

![image](https://user-images.githubusercontent.com/481448/148542615-d4569f7e-1412-45d9-801f-50bc34cd81bb.png)

Fluke 289 multimeter does this too:

![image](https://user-images.githubusercontent.com/481448/148542726-58cd19aa-674b-40ed-9e08-68440ba75675.png)

Having, a few buttons at the bottom and the sides would open up many options for users interacting with the device. As I'm not sure what sort of displays you are planning to use, I was actually in the middle of looking your different repos to see if I could figure it out when you replied :)

My enclosure is probably far bigger than what most MPG pendants opt for, but something like this enclosure might suit the 5x5 keypad and the MPG wheel, with room for a few soft buttons too:

https://www.thingiverse.com/thing:2218467

**Comment by mundsen** [comment]
I'm going to start on mine (solder the smd components on the pcb..  Anyone have one up and running that want to share images.. for inspiration?

**Comment by mundsen** [comment]
> It is likely that I will change the D socket to a smaller version or replacing it with different connector(s). The latest version of the firmware I am working on can work with only four wires, or soon maybe even two (for power, with wireless transmission of data) or none (if battery operated).
> 
> What is more interesting is a discussion about how to connect to the controller and which kind of enclosures could be used. From that a more appropriate PCB design can be made.
> 
> FYI I made a 5x5 keypad yesterday for testing, and I have more than 30 commands assigned already. It seems like this can be a powerful/flexible platform to build upon...

in the grblhal2000 Jog2K pendant project, they use an USB-C fron the pendant to the controller

**Comment by terjeio** [comment]
> in the grblhal2000 Jog2K pendant project, they use an USB-C fron the pendant to the controller

If I am not mistaken a Pi Zero acts as a USB host to UART comms translator between the controller and the pendant.

**Comment by andrewmarles** [comment]
> r

No, it is more simple than that.  The Jog2K uses a just a Pi Pico to emulate an I2C keypad to GRBLHAL.  The USB-C connector is just a physical interface that is convenient because it is compact, surface mounted and has enough conductors for KPSTR and e-stop as well as differential I2C (in the original implementation).  All of the communication is I2C, including the addition of DRO and status display.

I have not created a PR for the keypad code as it is kind of its own thing for the jog2k and Hal2K since the hardware is slightly bespoke.  But I am happy to formwat it as such if you are interested to take a look  Also I have plans to find a way to run a modified version your own MPG/DRO firmware on the jogger instead of my current code as I would prefer that the two projects converge rather than diverge, but sadly time is not infinite so I have not gotten to that yet.

**Comment by andrewmarles** [comment]
Having said all that, I went ahead and created the PR in case it is useful.
